File: PatientPortal/src/app/patientChart/models/medicationItemInfo.ts
export class MedicationItemInfo {
    medicationNameId: string;
    route: string;
    strength: string;
    unit: string;
    dosageForm: string;

    constructor(medicationNameId: string = "",
        route: string = "", dose = "", dosageForm: string = "", units: string = "") {
        this.medicationNameId = medicationNameId;
        this.route = route;
        this.strength = dose;
        this.unit = units;
        this.dosageForm = dosageForm;
    }
}

File: medico/Domain/Medico.Domain/Models/MedicationItemInfo.cs
using System;

namespace Medico.Domain.Models
{
    public class MedicationItemInfo
    {
        public Guid MedicationNameId { get; set; }

        public MedicationName MedicationName { get; set; }

        public string Route { get; set; }

        public string Strength { get; set; }

        public string Unit { get; set; }

        public string DosageForm { get; set; }
    }
}